eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentInitiator.scala

```diff
@@ -62,6 +78,18 @@ object PaymentInitiator {
                                 externalId: Option[String] = None,
                                 predefinedRoute: Seq[PublicKey] = Nil,
                                 assistedRoutes: Seq[Seq[ExtraHop]] = Nil,
-                                routeParams: Option[RouteParams] = None)
+                                routeParams: Option[RouteParams] = None) {
+    // We add one block in order to not have our htlcs fail when a new block has just been found.
+    def finalExpiry(currentBlockHeight: Long) = finalExpiryDelta.toCltvExpiry(currentBlockHeight + 1)
+  }
+
+  case class SendPaymentConfig(id: UUID,
```

Comment:
IMO this would just defer that code to somewhere else. Since in the DB we want to have two separate columns for those, if we use an Option here or a trait, the DB will have to know that and do the proper mapping when reading from the table...WDYT?